Network/Mail/SMTP.hs

```diff
@@ -309,27 +313,16 @@ simpleMail from to cc bcc subject parts =
 
 -- | Construct a plain text 'Part'
 plainTextPart :: TL.Text -> Part
-plainTextPart = Part "text/plain; charset=utf-8"
-              QuotedPrintableText Nothing [] . TL.encodeUtf8
+plainTextPart body = Part "text/plain; charset=utf-8"
+              QuotedPrintableText DefaultDisposition [] (PartContent $ TL.encodeUtf8 body)
 
 -- | Construct an html 'Part'
 htmlPart :: TL.Text -> Part
-htmlPart = Part "text/html; charset=utf-8"
-             QuotedPrintableText Nothing [] . TL.encodeUtf8
-
--- | Construct a file attachment 'Part'
-filePart :: T.Text -- ^ content type
-         -> FilePath -- ^ path to file
-         -> IO Part
-filePart ct fp = do
-    content <- BL.readFile fp
-    return $ Part ct Base64 (Just $ T.pack (takeFileName fp)) [] content
+htmlPart body = Part "text/html; charset=utf-8"
+             QuotedPrintableText DefaultDisposition [] (PartContent $ TL.encodeUtf8 body)
 
 lazyToStrict :: BL.ByteString -> B.ByteString
 lazyToStrict = B.concat . BL.toChunks
 
-crlf :: B8.ByteString
-crlf = B8.pack "\r\n"
-
-bsPutCrLf :: Handle -> ByteString -> IO ()
-bsPutCrLf h s = B8.hPut h s >> B8.hPut h crlf >> hFlush h
+bsPutCrLf :: Conn.Connection -> ByteString -> IO ()
+bsPutCrLf = Conn.connectionPut
```

Comment:
I don't think this preserves the behavior fo the original implementation. I can't see anything in the source of `connectionPut` that appends a crlf.